src/modules/email-microservice/domain/email-microservice-event-handler.ts

```diff
@@ -74,12 +84,52 @@ export class EmailMicroserviceEventHandler {
         });
     }
 
-    private async getEmailServiceProviderId(
-        allKontexteForPerson: PersonenkontextEventKontextData[],
-    ): Promise<string | undefined> {
-        // Array to hold the role IDs
-        const rollenIds: string[] = allKontexteForPerson.map((pk: PersonenkontextEventKontextData) => pk.rolleId);
+    @KafkaEventHandler(KafkaPersonRenamedEvent)
+    @EventHandler(PersonRenamedEvent)
+    @EnsureRequestContext()
+    public async handlePersonRenamedEvent(event: PersonRenamedEvent | KafkaPersonRenamedEvent): Promise<void> {
+        this.logger.info(
+            `Received PersonRenamedEvent, personId:${event.personId}, username:${event.username}, oldUsername:${event.oldUsername}`,
+        );
+
+        if (!this.emailResolverService.shouldUseEmailMicroservice()) {
+            this.logger.info(`Ignoring Event for personId:${event.personId} because email microservice is disabled`);
+            return;
+        }
+        if (!event.username) {
+            throw new Error(`Person with id:${event.personId} has no username, cannot resolve email.`);
+        }
+
+        const allKontexteForPerson: KontextWithOrgaAndRolle[] =
+            await this.personenkontextRepo.findByPersonWithOrgaAndRolle(event.personId);
+
+        const kennungen: string[] = allKontexteForPerson
+            .map((kontext: KontextWithOrgaAndRolle) => kontext.organisation.kennung)
+            .filter((kennung: string | undefined): kennung is string => !!kennung);
+        const uniqueKennungen: string[] = Array.from(new Set(kennungen));
```

Comment:
Unless there is some reason not to?
```suggestion
        const uniqueKennungen: string[] = uniq(kennungen);
```

Reply:
Snuck in while copying, changed